### PR TITLE
[12.0][IMP] cooperator: backport changes from migration to 14.0

### LINKED ISSUE #288: remove "subscription" from subscription.request.type selection
In subscription request form, `type` field has three option : 
1. subscription
2. new cooperator
3. capital increase

Check `subscription` in indeed useless and remove it from selection.

### PR DESCRIPTION
**Pull request by huguesdk**
also includes changes from #370.

## changes

*   fix mail templates: mostly whitespace cleanup, but also fix firstname reference for companies (instead of displaying “false”).
*   remove subscription request `subscription` state (use `new` instead) (was used when creating a subscription request from a partner).
*   fix partner matching by email or company register number when creating a subscription request.
*   fix subscription request name for companies: use company name (like previously) instead of representative name. the name is displayed in the title of the form, and can be used to search.
*   make country and language fields mandatory on the web form.
*   fix name of reports and attachments.
*   some code refactoring and cleanup.
*   improve `fr` and `fr_be` translations.

edit : fix https://github.com/coopiteasy/vertical-cooperative/issues/288

### CODE REVIEW

File: cooperator/models/partner.py
Comment (victor-champonnois) [inline cooperator/models/partner.py:89]: Why this change ? because effective_date is already False by default ?

File: cooperator/models/partner.py
Comment (victor-champonnois) [inline cooperator/models/partner.py:132]: Effective member is the same thing as effective cooperator ? This is misleading no ?

File: cooperator/models/partner.py
Comment (victor-champonnois) [inline cooperator/models/partner.py:144]: We could consider renaming this variable as "former_member", what do you think ? (this would not be done here, but we can create an issue for future komunigi work)

File: cooperator/models/partner.py
Comment (victor-champonnois) [inline cooperator/models/partner.py]: Why the duplication ? Maybe a comment would be welcome ?

File: cooperator/models/subscription_request.py
Comment (victor-champonnois) [inline cooperator/models/subscription_request.py:116]: I have the impression that self.search(pending_requests_domain) could return something even if the partner is not member.
And if the partner exists but is not member, the type is "new", not "increase" no ?

File: cooperator/models/subscription_request.py
Comment (victor-champonnois) [inline cooperator/models/subscription_request.py:152]: Wouldn't this overwrite the representative's name ?

**Review by victor-champonnois — APPROVED** [review]
LGTM, modulo some questions

File: cooperator/models/partner.py
Comment (huguesdk) [reply cooperator/models/partner.py:89]: because [compute functions must always set a value](https://github.com/OCA/maintainer-tools/wiki/Migration-to-version-13.0#tasks-to-do-in-the-migration) (in 13.0, but like this we keep differences between 12.0 and 14.0 to a minimum).

File: cooperator/models/partner.py
Comment (huguesdk) [reply cooperator/models/partner.py:132]: indeed! i actually didn’t read the string, just removed the quotes. we should settle on one, let’s ask the analysts.

File: cooperator/models/partner.py
Comment (huguesdk) [reply cooperator/models/partner.py:144]: would be better english, indeed. is there a document where all renaming suggestions are listed together? that  would be the place for this, as it would be part of a separate change.

File: cooperator/models/partner.py
Comment (victor-champonnois) [reply cooperator/models/partner.py:132]: tagging @polchampion on this

File: cooperator/models/partner.py
Comment (huguesdk) [reply cooperator/models/partner.py]: good point. this could also be written:
```python
if email:
    email = email.strip()
if not email:
    return self.browse()
```
or comments added to the current version:
```python
# email could be falsy
if not email:
    return self.browse()
email = email.strip()
# or could be only made of whitespace. this test must be done twice
# because None or False have no .strip() method.
if not email:
    return self.browse()
```
which version do you prefer?

File: cooperator/models/partner.py
Comment (victor-champonnois) [reply cooperator/models/partner.py:144]: Don't know about such a list. Maybe @robinkeunen knows where to keep track of those ?

File: cooperator/models/subscription_request.py
Comment (huguesdk) [reply cooperator/models/subscription_request.py:152]: yes, but the name of a subscription request for a company is supposed to be the name of the company (as it was in 9 and in 12 before the changes to the `name` field). this allows to search by the company name. `firstname` and `lastname` are still set to the name of the representative.

File: cooperator/models/partner.py
Comment (victor-champonnois) [reply cooperator/models/partner.py]: The second one is more helpful to the reader than the first one, and could prevent a future error.

File: cooperator/models/subscription_request.py
Comment (victor-champonnois) [reply cooperator/models/subscription_request.py:152]: OK :+1:

File: cooperator/models/partner.py
Comment (huguesdk) [reply cooperator/models/partner.py]: what about this?
```python
if email:
    email = email.strip()
# email could be falsy or be only made of whitespace.
if not email:
    return self.browse()
```

Comment (victor-champonnois) [reply cooperator/models/partner.py]: yes, even better :)

Comment (victor-champonnois) [reply cooperator/models/partner.py]: actually I still prefer the second one because it is easier to understand, but maybe that's just me ^^

Comment (huguesdk) [reply cooperator/models/partner.py]: @robinkeunen?

File: cooperator/models/partner.py
Comment (polchampion) [reply cooperator/models/partner.py:132]: @victor-champonnois "effective cooperator" sounds more adequate.

File: cooperator/models/subscription_request.py
Comment (huguesdk) [reply cooperator/models/subscription_request.py:116]: the logic i used is that if a partner has already made a subscription request (even not validated yet, but not canceled), subsequent subscription requests should be of type `increase`. only the first subscription request would be of type `new`. does that sound logical to you?

Comment (victor-champonnois) [reply cooperator/models/subscription_request.py:116]: OK, It is a good option. It is not ideal because there can be cases where the first request will be cancelled later, and the only request will have type "increase". But I guess this case is not very frequent and too costly to account for...?

Comment (polchampion) [reply cooperator/models/subscription_request.py:116]: Reading your comments, it looks like the type should ideally be attributed later, when the request is validated (not at creation), at which point it makes more sense to check whether the subscription is new or increase. Just saying, but probably ok to leave it as huguesdk proposes.

Comment (huguesdk) [reply cooperator/models/subscription_request.py:116]: very good remarks. the type on the request itself does not make much sense until it is validated. what is this type used for, anyway?

Comment (polchampion) [reply cooperator/models/subscription_request.py:116]: I'm not sure.

**Comment by huguesdk** [comment]
@cathLemb or @polchampion: could you please validate this functionally? it is deployed on the test servers (on all dbs, and should work on a new db also, of course). all changes are listed in the description.

File: cooperator/models/partner.py
Comment (victor-champonnois) [reply cooperator/models/partner.py:144]: issue opened here https://github.com/coopiteasy/vertical-cooperative/issues/386

File: cooperator/models/partner.py
Comment (victor-champonnois) [reply cooperator/models/partner.py:132]: @huguesdk do you want to fix this here or to create an issue ?

Comment (huguesdk) [reply cooperator/models/partner.py:132]: the string can be fixed here (and then also in #373), but if we want to change the field name, then we need an issue.

File: cooperator/models/partner.py
Comment (robinkeunen) [reply cooperator/models/partner.py:144]: No existing list of possible renamings no. Opening issues is the way to go.

Also, better English yet would be :

> Check this box if this cooperator is not an effective member anymore.

File: cooperator/models/partner.py
Comment (robinkeunen) [reply cooperator/models/partner.py]: I like the last version, easier on the eye, less ways out of the function.

Comment (carmenbianca) [reply cooperator/models/partner.py]: The version with the fewest returns I like best.

**Comment by cathLemb** [comment]
Use emc12-demo_test, emc12-demo-testFR and emc12-demo-testCH (new db i've created and installed the modules)
I've done tests on the belgian db (except for tax shelter). Everything seems ok. 
take share for a company/a person - from back end or front end
selling a person share
transfering a person share

- fix mail templates : mostly whitespace cleanup, but also fix firstname reference for companies (instead of displaying “false”).
False is indeed replaced, but the request to pay email is not harmonized with the rest of the template (concerning the whitespaces). I can live with it but I think it's better to harmonize. 

- remove subscription request subscription state (use new instead) (was used when creating a subscription request from a partner).
ok

- make country and language fields mandatory on the web form.
Seems ok but I don't see any difference with what is in prod?

- fix name of reports and attachments.
reports works 

- some code refactoring and cleanup.
difficult to test but everything seems fine. 

- improve fr and fr_be translations.
ok

- fix partner matching by email or company register number when creating a subscription request.
not sure what was the problem and what do I have to test? 

- fix subscription request name for companies : use company name (like previously) instead of representative name. the name is displayed in the title of the form, and can be used to search.
not sure what this is about?

**Pull request merged**
